### DISCUSSION TITLE
Adding a new email address for commits only (dont allow login)

### DISCUSSION THREAD

**Discussion by sovetski** [community/community#86530] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,

We can add multiple email addresses which is very cool, but the problem, we can log in to our account by using them and the password, even if the email is not "primary".

How can I for example my work email address for commits etc. but not give the possibility to log in with this email address?

Thanks

**Comment by Aryma-f4** [comment]
You can use your work email address for commits on GitHub without giving the possibility to log in with this email address by using a noreply email address from GitHub. Here’s how you can do it:
1. Go to your GitHub settings.
2. Click on ```Emails```.
3. Unselect the checkbox for ```Keep my email addresses private```, then reselect it again

This should update the email address shown in the text below the text box to the format ```ID+username@users.noreply.github.com``` . You can use this noreply email address for commits you push from the command line.This way, your personal email address will be kept private.

Please note that GitHub uses the email address to link commits. If you specifically remove this from the commits, GitHub won’t link them. So, it’s important to use the ```noreply``` email address provided by GitHub.

Remember, these changes only affect future commits, not past ones. For past commits, you might need to modify the commit history. Be careful with this, as it can affect your repository’s history.

**Reply by sovetski** [reply]
> You can use your work email address for commits on GitHub without giving the possibility to log in with this email address by using a noreply email address from GitHub. Here’s how you can do it:
> 
> 1. Go to your GitHub settings.
> 
> 2. Click on ```Emails```.
> 
> 3. Unselect the checkbox for ```Keep my email addresses private```, then reselect it again
> 
> 
> 
> This should update the email address shown in the text below the text box to the format ```ID+username@users.noreply.github.com``` . You can use this noreply email address for commits you push from the command line.This way, your personal email address will be kept private.
> 
> 
> 
> Please note that GitHub uses the email address to link commits. If you specifically remove this from the commits, GitHub won’t link them. So, it’s important to use the ```noreply``` email address provided by GitHub.
> 
> 
> 
> Remember, these changes only affect future commits, not past ones. For past commits, you might need to modify the commit history. Be careful with this, as it can affect your repository’s history.

Hi, thanks but it is only for the new commits, I want it for existing commits with email myname@mycompany.com to keep older commits linked to my profile

**Reply by Aryma-f4** [reply]
Sorry for the late reply... You can change the email address for existing commits in Git using the git rebase command. Here’s how you can do it:

1. Open your terminal.

2. Navigate to your local repository.

3. Run the following command:
```
git rebase -i HEAD~N -x "git commit --amend --author='Author Name <email@address.com>' --no-edit"
```
Replace ```N``` with the number of commits you want to change, and replace ```Author Name <email@address.com>``` with your name and the email address you want to use.

Please note that this command will rewrite the history of your repository, which can cause issues if you’ve already pushed these commits to a remote repository. If you’ve already pushed these commits, you’ll need to force push your changes with ```git push --force```

Also, keep in mind that this command changes both the author and the committer of the commits. If you only want to change the author, you can use the ``` --author ```option with ```git commit --amend```

Remember to backup your repository before running these commands, as they can permanently alter your Git history. If you’re not comfortable with these steps, you might want to reach out to a colleague with more experience with Git or consider using a GUI Git client that provides a more user-friendly interface for these operations.

If the problem persists, you might want to reach out to the support team for further assistance. Please note that these instructions are for Git. If you’re using a different version control system, the exact steps might be different.

**Reply by sovetski** [reply]
Thank you again for your time, but it is not what I want to achieve.

I know that we can modify the git history etc. but I can not say to my company to do it...
What I need is just to be able to add an email address on GitHub for commits without the ability to log in with them, that is why I am doing a feature request here.
